sklearn/utils/extmath.py

```diff
@@ -839,18 +748,28 @@ def _incremental_mean_and_var(X, last_mean, last_variance, last_sample_count):
     # new = the current increment
     # updated = the aggregated stats
     last_sum = last_mean * last_sample_count
-    new_sum = _safe_accumulator_op(np.nansum, X, axis=0)
+    if sample_weight is not None:
+        new_sum = _safe_accumulator_op(np.nansum, X * sample_weight[:, None],
```

Comment:
No need for ``sample_weight[:, None]`` (numpy will braodcast the sample weights).
```suggestion
        new_sum = _safe_accumulator_op(np.nansum, X * sample_weight,
```

Reply:
@jeremiedbb are you sure? All the tests fail with this applied and errors
```
        if sample_weight is not None:
>           new_sum = _safe_accumulator_op(np.nansum, X * sample_weight,
                                           axis=0)
E           ValueError: operands could not be broadcast together with shapes (3,4) (3,)

````

Reply:
I tested against @jeremiedbb 's suggestions by using `test_incremental_weighted_mean_and_variance_simple`, and tests will fail.
So, I reverted these three commits for now.

Reply:
We reverted these changes for now as they simply don't work

Reply:
Yep, my bad. I somehow thought that sample_weight had shape (n_samples, 1)